rfc_samplers.md

```diff
@@ -0,0 +1,125 @@
+
+Feature space
+-------------
+
+At a high level, a sampler returns a sequence of "clips" where a clip is a batch
+of (non-necessarily contiguous) frames. Based on an analysis of the sampling
+options of other libraries (torchvision, torchmultimodal, and various internal
+libraries), we want to support the following use-cases:
+
+
+**Clip location** (and number thereof)
+
+- Get `num_clips_per_video` random clips: typically called "Random" sampler
+- Get `num_clips_per_video` equally-spaced clips: "Uniform" sampler. Space is
+  index-based.
+- Get `num_clips_per_second` clips per seconds [within the [`start_seconds`,
+  `stop_seconds`] sampling range]: "Periodic" sampler.
+
+**Clip content**
+
+- Clip must be `num_frames_per_clip` frames spaced by `dilation` frames.
```

Comment:
Does it by default do uniform sampling of `num_frames_per_clip` if dilation is None?

Here's Llama4's current use case:
1. `num_clips_per_video` = 1, dilation=None, `num_frames_per_clip` will be uniformly sampled from the video.
2. `num_clips_per_video` > 1, dilation=None, a list of (start, end) timestamps are required to sample `num_clips_per_video`. Each clip then will have `num_frames_per_clip` uniformly sampled from it. If `num_frames_per_clip` > actual number of frames in any clip, we repeat the frames in order until we get desired num of frames (eg. actual frames = 3, need 4; we'll do (f1, f1, f2, f3); actual frames = 3, need 8; we'll do (f1, f1, f1, f2, f2, f2, f3, f3,) OR something similar).